api/v1alpha1/dataset_methods.go

```diff
@@ -0,0 +1,20 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package v1alpha1
+
+func (d *Dataset) Conf() *DatasetConf {
+	return d.Spec.Dataset
+}
+
+func (d *Dataset) GetStatus() *DatasetStatus {
+	return d.Status
+}
```

Comment:
Like make `Status` a field of the interface instead of having a `GetStatus()` method? Yeah I think that makes sense.

Reply:
Because I need to access the status inside some methods which can only access the interface but not the actual type, i.e. https://github.com/Alluxio/k8s-operator/blob/3c07724b3c807cafdb53918cf2463936592462a6/pkg/dataset/update_dataset_status.go#L29